components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/util/FrameworkUtils.java

```diff
@@ -1782,6 +1782,53 @@ public static String getIdpRoleClaimUri(ExternalIdPConfig externalIdPConfig) {
         return idpRoleClaimUri;
     }
 
+    /**
+     * Get the mapped URI for the IDP role mapping
+     * @param idpRoleClaimUri pass the IdpClaimUri created in getIdpRoleClaimUri method
+     * @param stepConfig Relevant stepConfig
+     * @param context Relevant authentication context
+     * @return idpRole claim uri in IDPs dialect or Custom dialect
+     */
+    public static String getMappedIdpRoleClaimUri(String idpRoleClaimUri, StepConfig stepConfig,
+                                                  AuthenticationContext context) {
+        // Set the incoming idpClaimUri.
+        // Finally return the incoming idpClaimUri if it is in expected dialect.
+        String idpRoleMappingURI = idpRoleClaimUri;
+
+        ApplicationAuthenticator authenticator = stepConfig.
+                getAuthenticatedAutenticator().getApplicationAuthenticator();
+
+        // Read the value from management console.
+        boolean useDefaultIdpDialect = context.getExternalIdP().useDefaultLocalIdpDialect();
+
+        // Read value from file based configuration.
+        boolean useLocalClaimDialectForClaimMappings =
+                FileBasedConfigurationBuilder.getInstance().isCustomClaimMappingsForAuthenticatorsAllowed();
+
+        Map<String, String> carbonToStandardClaimMapping;
+
+        // Check whether to use the default dialect or custom dialect.
+        if (useDefaultIdpDialect || useLocalClaimDialectForClaimMappings) {
+            String idPStandardDialect = authenticator.getClaimDialectURI();
+            try {
+                // Maps the idps dialect to standard dialect.
+                carbonToStandardClaimMapping = ClaimMetadataHandler.getInstance()
+                        .getMappingsMapFromOtherDialectToCarbon(idPStandardDialect, null,
+                                context.getTenantDomain(), false);
+                // check for role claim uri in the idaps dialect.
+                for (Entry<String, String> entry : carbonToStandardClaimMapping.entrySet()) {
+                    if (idpRoleMappingURI.equalsIgnoreCase(entry.getValue())) {
+                        idpRoleMappingURI = entry.getKey();
+                    }
+                }
+            } catch (ClaimMetadataException e) {
+                log.debug("Error in getting the mapping between idps and standard dialect");
```

Comment:
Let's say error in getting the mapping. And idpRoleMappingURI itself is returned. better to print that URI value in the logs